cli/installer/install-azd.ps1

```diff
@@ -315,106 +328,43 @@ try {
+                    Write-Error "Signature of $releaseArtifactFilename is not valid"
                     reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
                     exit 1
                 }
+            } catch {
+                Write-Error "Could not verify signature of $releaseArtifactFilename"
+                Write-Error $_
+                reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
+                exit 1
             }
+        }
 
-            Write-Verbose "Installing azd in $InstallFolder" -Verbose:$Verbose
-            if (!(Test-Path $InstallFolder)) {
-                New-Item -ItemType Directory -Path $InstallFolder -Force | Out-Null
-            }
-            $outputFilename = "$InstallFolder/azd"
-            test -w "$InstallFolder/"
-            if ($LASTEXITCODE) {
-                Write-Host "Writing to $InstallFolder/ requires elevated permission. You may be prompted to enter credentials."
-                sudo cp "$tempFolder/decompress/$binFilename" $outputFilename
-                if ($LASTEXITCODE) {
-                    Write-Error "Could not copy $tempfolder/decompress/$binFilename to $outputFilename"
-                    reportTelemetryIfEnabled 'InstallFailed' 'FileCopyFailed'
-                    exit 1
-                }
-            } else {
-                Copy-Item "$tempFolder/decompress/$binFilename" $outputFilename  -ErrorAction Stop | Out-Null
-            }
-        } else {
-            if (!$SkipVerify) {
-                try {
-                    Write-Verbose "Verifying signature of $releaseArtifactFilename" -Verbose:$Verbose
-                    $signature = Get-AuthenticodeSignature $releaseArtifactFilename
-                    if ($signature.Status -ne 'Valid') {
-                        Write-Error "Signature of $releaseArtifactFilename is not valid"
-                        reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
-                        exit 1
-                    }
-                } catch {
-                    Write-Error "Could not verify signature of $releaseArtifactFilename"
-                    Write-Error $_
-                    reportTelemetryIfEnabled 'InstallFailed' 'SignatureVerificationFailed'
-                    exit 1
-                }
+        Write-Verbose "Installing MSI" -Verbose:$Verbose
+        $MSIEXEC = "${env:SystemRoot}\System32\msiexec.exe"
+        $installProcess = Start-Process $MSIEXEC `
+            -ArgumentList @("/i", $releaseArtifactFilename, "/qn", "INSTALLDIR=`"$InstallFolder`"", "INSTALLEDBY=`"install-azd.ps1`"") `
+            -PassThru `
+            -Wait
+
+        if ($installProcess.ExitCode) {
+            if ($installProcess.ExitCode -eq 1603) {
+                Write-Host "A later verison of Azure Developer CLI is already installed. Use 'Add or remove programs' to uninstall the current version and try again."
```

Comment:
1603 could be *anything*. It's a generic exit code. Might as well be 1. I wouldn't assume as much. You could keep the second sentence, but the first sentence could very well be misleading. Could just say an error occurred, and they can try again - even after removing the old.